toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/FeatureFormState.kt

```diff
@@ -0,0 +1,487 @@
+import com.arcgismaps.mapping.featureforms.FormGroupState
+import com.arcgismaps.mapping.featureforms.FormInputNoValueOption
+import com.arcgismaps.mapping.featureforms.GroupFormElement
+import com.arcgismaps.mapping.featureforms.RadioButtonsFormInput
+import com.arcgismaps.mapping.featureforms.SwitchFormInput
+import com.arcgismaps.mapping.featureforms.TextAreaFormInput
+import com.arcgismaps.mapping.featureforms.TextBoxFormInput
+import com.arcgismaps.mapping.featureforms.TextFormElement
+import com.arcgismaps.mapping.featureforms.UtilityAssociationsFormElement
+import com.arcgismaps.toolkit.featureforms.internal.components.attachment.AttachmentElementState
+import com.arcgismaps.toolkit.featureforms.internal.components.barcode.BarcodeFieldProperties
+import com.arcgismaps.toolkit.featureforms.internal.components.barcode.BarcodeTextFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.base.BaseFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.base.BaseGroupState
+import com.arcgismaps.toolkit.featureforms.internal.components.base.FormStateCollection
+import com.arcgismaps.toolkit.featureforms.internal.components.base.MutableFormStateCollection
+import com.arcgismaps.toolkit.featureforms.internal.components.base.formattedValueAsStateFlow
+import com.arcgismaps.toolkit.featureforms.internal.components.base.mapValidationErrors
+import com.arcgismaps.toolkit.featureforms.internal.components.base.mapValueAsStateFlow
+import com.arcgismaps.toolkit.featureforms.internal.components.codedvalue.CodedValueFieldProperties
+import com.arcgismaps.toolkit.featureforms.internal.components.codedvalue.ComboBoxFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.codedvalue.RadioButtonFieldProperties
+import com.arcgismaps.toolkit.featureforms.internal.components.codedvalue.RadioButtonFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.codedvalue.SwitchFieldProperties
+import com.arcgismaps.toolkit.featureforms.internal.components.codedvalue.SwitchFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.datetime.DateTimeFieldProperties
+import com.arcgismaps.toolkit.featureforms.internal.components.datetime.DateTimeFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.text.FormTextFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.text.TextFieldProperties
+import com.arcgismaps.toolkit.featureforms.internal.components.text.TextFormElementState
+import com.arcgismaps.toolkit.featureforms.internal.components.utilitynetwork.UtilityAssociationsElementState
+import com.arcgismaps.toolkit.featureforms.internal.utils.fieldIsNullable
+import com.arcgismaps.toolkit.featureforms.internal.utils.toMap
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * The state object for a [FeatureForm] used by the [FeatureForm] composable. This class is
```

Comment:
linking to composable is broken as it takes me to the `FeatureForm` class instead
```suggestion
 * The state object for a [FeatureForm] used by the FeatureForm composable. This class is
```